misaligned.cpp

```diff
@@ -1,21 +1,44 @@
 #include <iostream>
 #include <assert.h>
 
+using namespace std;
+
+const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+
+string indexToColorPair(int majorColorIndex, int minorColorIndex){
+    string separator = " | ";
+    return to_string(majorColorIndex*5+minorColorIndex) + \
+                            separator + majorColor[majorColorIndex] + \
+                            separator + minorColor[minorColorIndex]; 
+}
+
 int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
     int i = 0, j = 0;
-    for(i = 0; i < 5; i++) {
-        for(j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+    for(; i < 5; i++) {
+        for(int j = 0; j < 5; j++) {
+            cout << indexToColorPair(i, j) << endl; 
         }
     }
     return i * j;
 }
 
 int main() {
-    int result = printColorMap();
-    assert(result == 25);
-    std::cout << "All is well (maybe!)\n";
+    printColorMap();
+    
+    int i = 0, j = 0;
+    for(; i < 5; i++){
+        for(; j < 5; j++){
+            cout << "Assertion for major color: " << i << " minor color: " << j \
+                 << " results in: " << endl;
+            string result = indexToColorPair(i, j);

```

Comment:
Good try, with the value check within output text.
But we must also check the alignment of the output string.
Please add test cases for alignment check too.